src/communication/v2/listener.hpp

```diff
@@ -53,28 +55,30 @@ class Listener final : public std::enable_shared_from_this<Listener<TSession, TS
 
   void Start() { DoAccept(); }
 
+  bool IsRunning() const noexcept { return alive_.load(std::memory_order_relaxed); }
+
  private:
-  Listener(IOContextPool &io_context_pool, TSessionData *data, ServerContext *server_context, tcp::endpoint &endpoint,
-           const std::string_view service_name, const int inactivity_timeout_sec)
-      : io_context_pool_(io_context_pool),
+  Listener(IOContextThreadPool &io_thread_context_pool, TSessionData *data, ServerContext *server_context,
```

Comment:
As `Listener` doesn't actually use the the functionalities of the thread pool, I think passing in an `io_context` reference is enough, that is what the `Listener` actually need.